packages/react-client/src/DeviceManager.ts

```diff
@@ -130,22 +130,24 @@ export class DeviceManager
     this.storageConfig?.saveLastDevice(info);
   }
 
-  // todo `audioDeviceId / videoDeviceId === true` means use last device
-  public async start(deviceId?: string | boolean) {
-    const shouldRestart = !!deviceId && deviceId !== this.deviceState.media?.deviceInfo?.deviceId;
+  public async start(deviceId?: string) {
+    const newDeviceId = deviceId ?? this.getLastDevice()?.deviceId ?? true; // true means start any available device
+    const currentDeviceId = this.deviceState.media?.deviceInfo?.deviceId;
 
-    const newDevice = deviceId === true ? this.getLastDevice()?.deviceId || true : deviceId;
+    const shouldReplaceDevice = Boolean(currentDeviceId && currentDeviceId !== newDeviceId);
+    const isDeviceStopped = currentDeviceId === undefined;
+    const shouldProceed = isDeviceStopped || shouldReplaceDevice;
 
     const trackConstraints = this.constraints;
 
-    const exactConstraints = shouldRestart && prepareMediaTrackConstraints(newDevice, trackConstraints);
+    const exactConstraints = shouldProceed && prepareMediaTrackConstraints(newDeviceId, trackConstraints);
     if (!exactConstraints) return;
 
     this.deviceState.mediaStatus = "Requesting";
 
     this.emit(
       "devicesStarted",
-      { ...this.deviceState, restarting: shouldRestart, constraints: newDevice },
+      { ...this.deviceState, restarting: shouldReplaceDevice, constraints: newDeviceId },
```

Comment:
Why we are passing `deviceId` as `constraints`?

Reply:
Good catch, fixed!